Ok so we are doing a binary classification model, and we can do logisitic regression, naive bayes, decision tree, random forest, or XGBoost models.

# Step 0: Import Packages

In [5]:
# Operational Packages
import numpy as np # Numerical calculations
import pandas as pd # Data handling

# Visualization Packages
import matplotlib.pyplot as plt # Lower-level graphics
import seaborn as sns # High-level graphics

# General Machine Learning
from sklearn.model_selection import train_test_split, GridSearchCV, PredefinedSplit # Train test split for model building and Grid Search for tuning
from sklearn.metrics import accuracy_score, precision_score, recall_score, f1_score # Model Performance metrics
from sklearn.metrics import ConfusionMatrixDisplay, confusion_matrix # Confusion Matrix

### Machine Learning Models

# Modeling and evaluation for K-means
from sklearn.cluster import KMeans
from sklearn.metrics import silhouette_score 
from sklearn.preprocessing import StandardScaler # Scaling data 

# Decision Tree
from sklearn.tree import DecisionTreeClassifier # Decision Tree
# This function displays the splits of the tree
from sklearn.tree import plot_tree 

# Random Forest 
from sklearn.ensemble import RandomForestClassifier # Random Forest Classifier

# XGBoost 
from xgboost import XGBClassifier
# This is the function that helps plot feature importance 
from xgboost import plot_importance


# For saving Machine Learning Models once we fit them
import pickle

# Step 1: Examine Data 

In [6]:
# Load the dataset into a pandas dataframe
df = pd.read_csv('train.csv')
df.head()

,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
0,1,0,3,"Braund, Mr. Owen Harris",male,22.0,1,0,A/5 21171,7.2500,NaN,S
1,2,1,1,"Cumings, Mrs. John Bradley (Florence Briggs Th...",female,38.0,1,0,PC 17599,71.2833,C85,C
2,3,1,3,"Heikkinen, Miss. Laina",female,26.0,0,0,STON/O2. 3101282,7.9250,NaN,S
3,4,1,1,"Futrelle, Mrs. Jacques Heath (Lily May Peel)",female,35.0,1,0,113803,53.1000,C123,S
4,5,0,3,"Allen, Mr. William Henry",male,35.0,0,0,373450,8.0500,NaN,S


In [9]:
#examine the shape of the dataset
df.shape

(891, 12)

The dataset contains 891 rows and 12 columns, so 891 passengers with 12 chararcteristics describing each one of them.

In [10]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    object 
 4   Sex          891 non-null    object 
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    object 
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    object 
 11  Embarked     889 non-null    object 
dtypes: float64(2), int64(5), object(5)
memory usage: 83.7+ KB


In [11]:
df.describe()

,PassengerId,Survived,Pclass,Age,SibSp,Parch,Fare
count,891.000000,891.000000,891.000000,714.000000,891.000000,891.000000,891.000000
mean,446.000000,0.383838,2.308642,29.699118,0.523008,0.381594,32.204208
std,257.353842,0.486592,0.836071,14.526497,1.102743,0.806057,49.693429
min,1.000000,0.000000,1.000000,0.420000,0.000000,0.000000,0.000000
25%,223.500000,0.000000,2.000000,20.125000,0.000000,0.000000,7.910400
50%,446.000000,0.000000,3.000000,28.000000,0.000000,0.000000,14.454200
75%,668.500000,1.000000,3.000000,38.000000,1.000000,0.000000,31.000000
max,891.000000,1.000000,3.000000,80.000000,8.000000,6.000000,512.329200


some initial concerns about the data is that is that there are over 100 entries missing their age and around 700 entries are missing a cabin number, I will have to investigate that.

# Step 2: Exploratory Data Analysis (EDA)

# Step 3: Feature Engineering

# Step 4: Split the data

## Step 5: Model Building

# Step 6: Model Evaluation